tests/valid/Fail_Valid_1.whiley

```diff
@@ -0,0 +1,16 @@
+type nat is (int x) where x >= 0
+type neg is (int x) where x < 0
+
+function f(int|null x) -> bool|null:
+    //
+    if x is nat:
+        return true
+    else if x is neg:
+        return false
+    else:
+        fail
```

Comment:
This code is taken from example in the spec accompanying `fail` statement. The compiler reports it as valid, which is good because it matches the spec, but I would actually expect it to fail verification because it can reach the `fail` statement if the input is `null`. Am I misunderstanding something?